### PR TITLE
Fix markdown support

### LINKED ISSUE #6307: Support markdown formatting in Gitcoin Grants posts in the Activity tab
It would be great if markdown formatting were supported in Gitcoin Grants posts in the Activity tab. Right now, it ends up looking odd if you try to use multiple newlines and hyperlinks in a post.

Screenshot 2020-03-26 12 03 04

### PR DESCRIPTION
**Pull request by iamonuwa**
##### Description
Add markdown support to Kudos, Activity, Profile and Grants 

##### Refers/Fixes
Closes #6307 

##### Testing

https://www.loom.com/share/0960f2a7c9544e88bcc3623c791300ee

### CODE REVIEW

**Comment by iamonuwa** [comment]
@thelostone-mc had to close the old PR and open this one.

File: app/grants/templatetags/markdown.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+ 
+from django import template
+import mistune
+from pygments import highlight
+from pygments.lexers import get_lexer_by_name
+from pygments.formatters import HtmlFormatter
+
+register = template.Library()
+
+class HighlightRenderer(mistune.Renderer):
```
Comment (thelostone-mc) [inline app/grants/templatetags/markdown.py:10]: Travis fails with this 

```
 class HighlightRenderer(mistune.Renderer):
E   AttributeError: module 'mistune' has no attribute 'Renderer'
```

File: requirements/base.txt [hunk 2]
```diff
@@ -94,4 +94,5 @@ pandas
 wiki
 django-bulk-update
 elasticsearch
-
+Pygments==2.1.3
```
Comment (thelostone-mc) [inline requirements/base.txt:97]: CI failures 

`pkg_resources.VersionConflict: (Pygments 2.1.3 (/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages), Requirement.parse('Pygments>=2.2'))`

Could we update this and see if it still works as expected ?

Comment (iamonuwa) [reply requirements/base.txt:97]: Alright.

**Comment by octavioamu** [comment]
Hey @iamonuwa can you post a video showing how it works with mention tagging and automatic links? We parse all the text in JS to check if there is a url then convert it into a html link, the same with mentions we create profile/username. Im worry with this change the links get converted 2 time since the js will see `<a href="link">link</a>` and probably will try to convert the href content becaming  `<a href="<a href="link">link</a>">link</a>`

**Comment by iamonuwa** [comment]
Alright, coming through

**Comment by iamonuwa** [comment]
@octavioamu apologies for the delay. I've successfully loaded up the app. I'll be sending in the proof today.

**Comment by iamonuwa** [comment]
@octavioamu here you go https://www.loom.com/share/074e8b67ad8f42aabf0fa453938f5de7. Apologies for the delay

File: app/retail/templates/shared/activity.html [hunk 3]
```diff
@@ -193,18 +192,23 @@
...
           </a>
         {% elif row.activity_type == 'status_update' or  row.activity_type == 'silent_status_update' %}
+
           {% if row.metadata.resource.type != 'background' %}
-            <span>{{ row.metadata.title }} {% if row.metadata.ask != 'undefined' %}{{ row.metadata.ask }}{%endif%}</span>
+            <span>{{ row.metadata.title | markdown | safe }} {% if row.metadata.ask != 'undefined' %}{{ row.metadata.ask }}{%endif%}</span>
 
             {% if row.metadata.video and row.metadata.gfx != 'undefined' %}
-            <div id="{% if row.metadata.roomname %}{{row.metadata.roomname}}{%else%}meet{{row.pk}}{%endif%}" class="click_here_to_join_video" data-roomname="{% if row.metadata.roomname %}{{row.metadata.roomname}}{%else%}meet{{row.pk}}{%endif%}" data-src='/static/v2/images/{{row.metadata.gfx}}'><img src='/static/v2/images/{{row.metadata.gfx}}'></div>
+              <div id="{% if row.metadata.roomname %}{{row.metadata.roomname}}{%else%}meet{{row.pk}}{%endif%}"
```
Comment (danlipert) [inline app/retail/templates/shared/activity.html:200]: what was changed here?

File: app/retail/templates/shared/activity.html [hunk 4]
```diff
@@ -193,18 +192,23 @@
...
+              <div id="{% if row.metadata.roomname %}{{row.metadata.roomname}}{%else%}meet{{row.pk}}{%endif%}" 
+                   class="click_here_to_join_video" data-roomname="{% if row.metadata.roomname %}{{row.metadata.roomname}}{%else%}meet{{row.pk}}{%endif%}" data-src='/static/v2/images/{{row.metadata.gfx}}'
+              >
+                <img src='/static/v2/images/{{row.metadata.gfx}}'>
+              </div>
 
-            <p class='num_participants m-0 p-0' data-roomname="+roomName+">
-              <p class='m-0 p-0 indie_chat_indicator {% if row.video_participants_count == 0 %}offline{%endif%}'>
-                •
+              <p class='num_participants m-0 p-0' data-roomname="+roomName+">
```
Comment (danlipert) [inline app/retail/templates/shared/activity.html:206]: Is this just a style change?

**Comment by iamonuwa** [comment]
Had to remove the sanitizer for the markdown to work. You can validate inputs sent to the server

File: app/retail/templates/shared/activity.html (hunk 3 above)
Comment (iamonuwa) [reply app/retail/templates/shared/activity.html:200]: No change occurred here.

File: app/retail/templates/shared/activity.html (hunk 4 above)
Comment (iamonuwa) [reply app/retail/templates/shared/activity.html:206]: Yes, same here

**Comment by iamonuwa** [comment]
@danlipert @octavioamu any update guys?

**Comment by thelostone-mc** [comment]
^ @danlipert + @octavioamu

**Comment by thelostone-mc** [comment]
Closing this out now as we don't need it ! will reopen if needed

**Pull request closed without merging**